File: server/core/model/variable_info.go
package model

import "time"

type VariableInfo struct {
	ID               int64      `gorm:"primaryKey;autoIncrement" json:"id"`
	FlowDefinitionID int64      `gorm:"column:flow_definition_id" json:"flowDefinitionId"`
	VariableKey      string     `gorm:"column:variable_key;size:30" json:"variableKey"`
	VariableName     string     `gorm:"column:variable_name;size:30" json:"variableName"`
	VariableType     int        `gorm:"column:variable_type" json:"variableType"`
	DataType         string     `gorm:"column:data_type;type:text" json:"dataType,omitempty"`
	CreatedAt        *time.Time `gorm:"autoCreateTime" json:"createdAt,omitempty"`
	CreatedBy        *int64     `json:"createdBy,omitempty"`
	UpdatedAt        *time.Time `gorm:"autoUpdateTime" json:"updatedAt,omitempty"`
	UpdatedBy        *int64     `json:"updatedBy,omitempty"`
}

func (VariableInfo) TableName() string { return "t_variable_info" }

File: server/src/main/java/com/javaoffers/nexus/core/model/VariableInfo.java
package com.javaoffers.nexus.core.model;

import com.javaoffers.brief.modelhelper.anno.BaseModel;
import com.javaoffers.brief.modelhelper.anno.BaseUnique;
import com.javaoffers.brief.modelhelper.anno.ColName;
import lombok.Data;

import java.util.Date;

/**
 * 注意：原 GORM 模型无 deleted 字段（硬删除），故此处不声明 IsDel。
 */
@BaseModel("t_variable_info")
@Data
public class VariableInfo {
    @BaseUnique
    private Long id;
    private Date createdAt;
    private Long createdBy;
    private Date updatedAt;
    private Long updatedBy;

    @ColName(value = "id", excludeColAll = true)
    private Long countId;

    private Long flowDefinitionId;
    private String variableKey;
    private String variableName;
    private Integer variableType;
    private String dataType;
}
